apps/studio/src/components/ResourceSelector/useResourceStack.tsx

```diff
@@ -0,0 +1,59 @@
+import { useCallback, useMemo, useState } from "react"
+
+import type { ResourceItemContent } from "~/schemas/resource"
+import { trpc } from "~/utils/trpc"
+
+export const useResourceStack = ({
+  siteId,
+  selectedResourceId,
+  existingResource,
+}: {
+  siteId: number
+  selectedResourceId: string | undefined
+  existingResource: ResourceItemContent | undefined
+}) => {
+  const [pendingMovedItemAncestryStack] =
+    trpc.resource.getAncestryWithSelf.useSuspenseQuery({
+      siteId: String(siteId),
+      resourceId: existingResource?.id,
+    })
+
+  // NOTE: This is the stack of user's navigation through the resource tree
+  // NOTE: We should always start the stack from `/` (root)
+  // so that the user will see a full overview of their site structure
+  const [resourceStack, setResourceStack] = useState<ResourceItemContent[]>(
+    pendingMovedItemAncestryStack,
+  )
+
+  const [isResourceHighlighted, setIsResourceHighlighted] =
+    useState<boolean>(!!selectedResourceId)
```

Comment:
nit: not entirely sure this should be here rather than letting the caller handle it